src/py21cmmc/mcmc.py

```diff
@@ -180,8 +180,144 @@ def run_mcmc(
+
+    Parameters
+    ----------
+    core_modules : list
+        A list of objects which define the necessary methods to be core modules (see :mod:`~py21cmmc.mcmc.core`).
+    likelihood_modules : list
+        A list of objects which define the necessary methods to be likelihood modules (see
+        :mod:`~py21cmmc.mcmc.likelihood`)
+    params : dict
+        Parameters which will be sampled by the chain. Each entry's key specifies the name of the parameter, and
+        its value is an iterable `(val, min, max, width)`, with `val` the initial guess, `min` and `max` the hard
+        boundaries on the parameter's value, and `width` determining the size of the initial ball of walker positions
+        for the parameter.
+    datadir : str, optional
+        Directory to which MCMC info will be written (eg. logs and chain files)
+    model_name : str, optional
+        Name of the model, which determines filenames of outputs.
+    log_level_mcmc : (int or str, optional)
+        The logging level of the cosmoHammer log file.
+        for details. This setting affects only the file output by the MCMC run.
+    log_level_mcmc_stream : (int or str, optional)
+        The logging level of the stdout/stderr stream from cosmoHammer. This has the same output as the cosmoHammer
+        log file, but is printed to screen. See `log_level_mcmc` for input specifications.
+    log_level_21CMMC : (int or str, optional)
+        The logging level of the 21cmFAST Python code (specifically the "21CMMC" logging object). By default, this
+        logger has only a stdout handler. See https://docs.python.org/3/library/logging.html#logging-levels
+
+    Essential MultiNest Parameters
+    ------------------------------
+    n_live_points : int, optional
+        number of live points
+    importance_nested_sampling : bool, optional
+        If True, Multinest will use Importance Nested Sampling (INS). Read http://arxiv.org/abs/1306.2144
+        for more details on INS. Please read the MultiNest README file before using the INS in MultiNest v3.0.
+    sampling_efficiency : float, optional
+        defines the sampling efficiency. 0.8 and 0.3 are recommended for parameter estimation & evidence evalutation
+    evidence_tolerance : float, optional
+        A value of 0.5 should give good enough accuracy.
+    max_iter : int, optional
+        maximum number of iterations. 0 is unlimited.
+    multimodal : bool, optional
+        whether or not to detect multi mode
+    write_output : bool, optional
+        write output files? This is required for analysis.
+    resume : bool, optional
+        If True, Multinest will read in from file the last locations of the iso-likelihood contours to "continue" 
+        the sampling if the sampler hasn't sufficently converged.
+
+    Other Parameters
+    ----------------
+    All other parameters are passed directly to multinest
+
+    Returns
+    -------
+    sampler : multinest sampler
+    """
+    import pymultinest
+
```

Comment:
From here to line 299 is very similar to `run_mcmc`, and we should probably refactor so we don't have redundant code.

Reply:
Are you suggesting simply renaming them or putting run_multinest as an option for run_mcmc?

Reply:
I think I would favour the latter, actually. Let's just add an option for `run_mcmc` like `sampler`, with `emcee` as the default, but one could specify `"multinest"` if they want.

Reply:
No, I think all of the new multinest options can be part of the existing `**mcmc_options`, and you can give them defaults in the function body by doing something like

```python
n_live_points = mcmc_options.get('n_live_points', 100)
```

Reply:
Also, 'resume' does the same thing as 'continue_sampling', so that's one less parameter.

Reply:
done. but we might need to consider renaming `run_mcmc` because multinest is not strictly speaking an mcmc.

Reply:
I think it's close enough that no-one is going to care :-)